packages/jupyterlab-gridstack/src/editor/gridstack/gridstackLayout.ts

```diff
@@ -234,7 +240,10 @@ export class GridStackLayout extends Layout {
     }
 
     this._gridItems.push(item);
+
+    MessageLoop.sendMessage(item, Widget.Msg.BeforeAttach);
     this._grid.addWidget(item.node, options);
+    MessageLoop.sendMessage(item, Widget.Msg.AfterAttach);
```

Comment:
Since the item is added by gridstack, these messages are not sent by Lumino like with `Widget.attach`, so we send them manually.